### PR TITLE
Clusters return complete url instead of DNS names

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :hammer_and_wrench: Description

Addresses https://github.com/hashicorp/terraform-provider-hcp/issues/124 and https://github.com/hashicorp/terraform-provider-hcp/issues/133. While it's possible in the future the backend responses will be updated to return a complete URL, for now the provider will handle the translation.

### :building_construction: Acceptance tests

✖️ Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccConsulCluster'

--- PASS: TestAccConsulCluster (551.46s)

$ make testacc TESTARGS='-run=TestAccVaultCluster'

--- PASS: TestAccVaultCluster (683.37s)
```

### CODE REVIEW

File: internal/provider/resource_consul_cluster.go [hunk 1]
```diff
@@ -450,12 +451,12 @@ func setConsulClusterResourceData(d *schema.ResourceData, cluster *consulmodels.
 	}
 
 	if publicEndpoint {
-		if err := d.Set("consul_public_endpoint_url", cluster.DNSNames.Public); err != nil {
+		if err := d.Set("consul_public_endpoint_url", fmt.Sprintf("https://%s:8200", cluster.DNSNames.Public)); err != nil {
```
Comment (riddhi89) [inline internal/provider/resource_consul_cluster.go:454]: Hmm, any reason the port has been included here? And which port is this 🤔

**Review by jfreda — COMMENTED** [review]
This looks good for Vault, but I think the Consul port needs to be changed (and `testAccCheckFullURL` refactored accordingly).

File: internal/provider/resource_consul_cluster.go (hunk 1 above)
Comment (bcmdarroch) [reply internal/provider/resource_consul_cluster.go:454]: This is the [port the UI sets](https://github.com/hashicorp/cloud-ui/blob/81c0cbed5a0b5b8211184f6e26891adf420c625f/engines/vault/addon-test-support/pages/components/page/detail.js#L7-L8) in the copy component on the cluster details page. Not sure where it came from originally though.

Comment (riddhi89) [reply internal/provider/resource_consul_cluster.go:454]: Hmm, I think dropping it altogether might be better. It keeps consistent with the HCP UI too?

Comment (riddhi89) [reply internal/provider/resource_consul_cluster.go:454]: This is what I get when I copy the url from the UI:
```https://consul-cluster-primary.consul.11eb5071-85f5-1eb2-992c-0242ac110003.aws.hashicorp.cloud
https://consul-cluster-primary.private.consul.11eb5071-85f5-1eb2-992c-0242ac110003.aws.hashicorp.cloud```
Does not look like there are any ports attached.

Comment (jfreda) [reply internal/provider/resource_consul_cluster.go:454]: Oh whoops, I think I reviewed at like the same time that @riddhi89 did yesterday and I missed this thread 🙂  HCP Vault uses Vault's default port 8200 for its API, but I believe that HCP Consul just uses port 443 for its API so its URL shouldn't need to include a port (but will let @riddhi89 confirm this for sure).

Comment (riddhi89) [reply internal/provider/resource_consul_cluster.go:454]: Yup. We can make Consul HTTP API requests using HTTPS/443 port:

Comment (bcmdarroch) [reply internal/provider/resource_consul_cluster.go:454]: Thanks! 🙌

**Review by riddhi89 — APPROVED** [review]

**Review by jfreda — APPROVED** [review]

**Pull request merged**